app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,61 @@
+class TasksController < ApplicationController
+
+  def index
+    @tasks = Task.all
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    task = Task.new(task_params)
+    if task.save
+      redirect_to tasks_path
+    end
+  end
+
+  def show
+    task_id = params[:id]
+    @task = Task.find(task_id)
+  end
+
+  def edit
+    @task = task.find(params[:id])
+  end
```

Comment:
On line 24, it should be `Task.find`, not `task.find`, since we're looking at the entire collection of tasks.